[BUG] node_disconnected_exception when trying to configure oci repository

### What is the bug?
_A clear and concise description of the bug._
node_disconnected_exception when trying to configure oci repository

### How can one reproduce the bug?
_Steps to reproduce the behavior._
I am trying to configure the repository with Instance Principal for OCI ObjectStorage plugin as defined in https://github.com/opensearch-project/opensearch-oci-object-storage. But when I try to configure with this command, I always get an error -
[opensearch@opensearch-es-data-0 ~]$ curl -XPUT "http://localhost:9200/_snapshot/oci_repository
" -H 'Content-Type: application/json' -d"
> {
> \"type\": \"oci\",
> \"settings\":

{ > \"client\": \"default\", > \"endpoint\": \"[https://objectstorage.us-ashburn-1.oraclecloud.com|https://objectstorage.us-ashburn-1.oraclecloud.com/] \", > \"bucket\": \"opensearch_backup\", > \"namespace\": \"-------\", > \"bucket_compartment_id\": \"NOT ACTUAL\", > \"useInstancePrincipal\": true, > \"forceBucketCreation\" : true > }
> }
> "

{"error":{"root_cause":[

{"type":"node_disconnected_exception","reason":"[opensearch-es-master-2][10.244.6.59:9300][cluster:admin/repository/put] disconnected"}
],"type":"master_not_discovered_exception","reason":"NodeDisconnectedException[[opensearch-es-master-2][10.244.6.59:9300][cluster:admin/repository/put] disconnected]","caused_by":{"type":"node_disconnected_exception","reason":"[opensearch-es-master-2][10.244.6.59:9300][cluster:admin/repository/put] disconnected"}},"status":503}
 
Cluster seems to be in healthy state before the above call -
curl -v -X GET
http://localhost:9200/_cluster/health?pretty

About to connect() to localhost port 9200 (#0)
Trying ::1...
Connected to localhost (::1) port 9200 (#0)
> GET /_cluster/health?pretty HTTP/1.1
> User-Agent: curl/7.29.0
> Host: localhost:9200
> Accept: /
>
< HTTP/1.1 200 OK
< content-type: application/json; charset=UTF-8
< content-length: 502
< { "cluster_name" : "opensearch-tracking", "status" : "green", "timed_out" : false, "number_of_nodes" : 6, "number_of_data_nodes" : 3, "discovered_master" : true, "active_primary_shards" : 0, "active_shards" : 0, "relocating_shards" : 0, "initializing_shards" : 0, "unassigned_shards" : 0, "delayed_unassigned_shards" : 0, "number_of_pending_tasks" : 0, "number_of_in_flight_fetch" : 0, "task_max_waiting_in_queue_millis" : 0, "active_shards_percent_as_number" : 100.0 }
 
### What is the expected behavior?
_A clear and concise description of what you expected to happen._

### What is your host/environment?
_Operating system, version._

### Do you have any screenshots?
_If applicable, add screenshots to help explain your problem._

### Do you have any additional context?
_Add any other context about the problem._

Here is the stack trace in the OpenSearch log associated with the above issue:

{"type": "server", "timestamp": "2022-11-16T01:21:44,344Z", "level": "WARN", "component": "o.o.r.RepositoriesService", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "failed to create repository [oci][oic_repository]", "cluster.uuid": "-ck8Z8_BT0Kmh84eaSZ0ig", "node.id": "_iqOPU1UQqyauFtucEVQ7A" ,
"stacktrace": ["java.lang.IllegalStateException: Client instance has been closed.",
"at org.glassfish.jersey.internal.guava.Preconditions.checkState(Preconditions.java:169) ~[?:?]",
"at org.glassfish.jersey.client.JerseyClient.checkNotClosed(JerseyClient.java:257) ~[?:?]",
"at org.glassfish.jersey.client.JerseyClient.target(JerseyClient.java:272) ~[?:?]",
"at org.glassfish.jersey.client.JerseyClient.target(JerseyClient.java:56) ~[?:?]",
"at com.oracle.bmc.auth.AbstractFederationClientAuthenticationDetailsProviderBuilder.simpleRetry(AbstractFederationClientAuthenticationDetailsProviderBuilder.java:454) ~[?:?]",
"at com.oracle.bmc.auth.AbstractFederationClientAuthenticationDetailsProviderBuilder.autoDetectEndpointUsingMetadataUrl(AbstractFederationClientAuthenticationDetailsProviderBuilder.java:247) ~[?:?]",
"at com.oracle.bmc.auth.AbstractFederationClientAuthenticationDetailsProviderBuilder.autoDetectUsingMetadataUrl(AbstractFederationClientAuthenticationDetailsProviderBuilder.java:233) ~[?:?]",
"at com.oracle.bmc.auth.InstancePrincipalsAuthenticationDetailsProvider$InstancePrincipalsAuthenticationDetailsProviderBuilder.build(InstancePrincipalsAuthenticationDetailsProvider.java:77) ~[?:?]",
"at org.opensearch.repositories.oci.OciObjectStorageClientSettings.toAuthDetailsProvider(OciObjectStorageClientSettings.java:144) ~[?:?]",
"at org.opensearch.repositories.oci.OciObjectStorageClientSettings.<init>(OciObjectStorageClientSettings.java:112) ~[?:?]",
"at org.opensearch.repositories.oci.OciObjectStorageRepository.<init>(OciObjectStorageRepository.java:116) ~[?:?]",
"at org.opensearch.repositories.oci.OciObjectStoragePlugin.lambda$getRepositories$0(OciObjectStoragePlugin.java:54) ~[?:?]",
"at org.opensearch.repositories.Repository$Factory.create(Repository.java:88) ~[opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.repositories.RepositoriesService.createRepository(RepositoriesService.java:563) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.repositories.RepositoriesService.registerRepository(RepositoriesService.java:177) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.action.admin.cluster.repositories.put.TransportPutRepositoryAction.masterOperation(TransportPutRepositoryAction.java:101) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.action.admin.cluster.repositories.put.TransportPutRepositoryAction.masterOperation(TransportPutRepositoryAction.java:55) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.action.support.master.TransportMasterNodeAction.masterOperation(TransportMasterNodeAction.java:124) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.lambda$doStart$3(TransportMasterNodeAction.java:197) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.action.ActionRunnable$2.doRun(ActionRunnable.java:86) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:50) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.common.util.concurrent.OpenSearchExecutors$DirectExecutorService.execute(OpenSearchExecutors.java:296) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.doStart(TransportMasterNodeAction.java:197) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.action.support.master.TransportMasterNodeAction.doExecute(TransportMasterNodeAction.java:140) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.action.support.master.TransportMasterNodeAction.doExecute(TransportMasterNodeAction.java:72) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:194) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.action.support.TransportAction.execute(TransportAction.java:169) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.action.support.TransportAction.execute(TransportAction.java:97) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.client.node.NodeClient.executeLocally(NodeClient.java:108) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.client.node.NodeClient.doExecute(NodeClient.java:95) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.client.support.AbstractClient.execute(AbstractClient.java:433) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.client.support.AbstractClient$ClusterAdmin.execute(AbstractClient.java:730) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.client.support.AbstractClient$ClusterAdmin.putRepository(AbstractClient.java:973) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.rest.action.admin.cluster.RestPutRepositoryAction.lambda$prepareRequest$0(RestPutRepositoryAction.java:75) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.rest.BaseRestHandler.handleRequest(BaseRestHandler.java:128) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.rest.RestController.dispatchRequest(RestController.java:306) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.rest.RestController.tryAllHandlers(RestController.java:392) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.rest.RestController.dispatchRequest(RestController.java:235) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.http.AbstractHttpServerTransport.dispatchRequest(AbstractHttpServerTransport.java:361) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.http.AbstractHttpServerTransport.handleIncomingRequest(AbstractHttpServerTransport.java:440) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.http.AbstractHttpServerTransport.incomingRequest(AbstractHttpServerTransport.java:351) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.http.netty4.Netty4HttpRequestHandler.channelRead0(Netty4HttpRequestHandler.java:55) [transport-netty4-client-1.2.4.jar:1.2.4]",
"at org.opensearch.http.netty4.Netty4HttpRequestHandler.channelRead0(Netty4HttpRequestHandler.java:41) [transport-netty4-client-1.2.4.jar:1.2.4]",
"at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at org.opensearch.http.netty4.Netty4HttpPipeliningHandler.channelRead(Netty4HttpPipeliningHandler.java:71) [transport-netty4-client-1.2.4.jar:1.2.4]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) [netty-codec-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) [netty-codec-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.handler.codec.MessageToMessageCodec.channelRead(MessageToMessageCodec.java:111) [netty-codec-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) [netty-codec-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) [netty-codec-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:324) [netty-codec-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296) [netty-codec-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286) [netty-handler-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) [netty-codec-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:722) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:623) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:586) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986) [netty-common-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.72.Final.jar:4.1.72.Final]",
"at java.lang.Thread.run(Thread.java:834) [?:?]"] }
{"type": "server", "timestamp": "2022-11-16T01:21:44,347Z", "level": "WARN", "component": "r.suppressed", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "path: /_snapshot/oic_repository, params: {repository=oic_repository}", "cluster.uuid": "-ck8Z8_BT0Kmh84eaSZ0ig", "node.id": "_iqOPU1UQqyauFtucEVQ7A" ,
"stacktrace": ["org.opensearch.repositories.RepositoryException: [oic_repository] failed to create repository",
"at org.opensearch.repositories.RepositoriesService.createRepository(RepositoriesService.java:572) ~[opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.repositories.RepositoriesService.registerRepository(RepositoriesService.java:177) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.action.admin.cluster.repositories.put.TransportPutRepositoryAction.masterOperation(TransportPutRepositoryAction.java:101) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.action.admin.cluster.repositories.put.TransportPutRepositoryAction.masterOperation(TransportPutRepositoryAction.java:55) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.action.support.master.TransportMasterNodeAction.masterOperation(TransportMasterNodeAction.java:124) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.lambda$doStart$3(TransportMasterNodeAction.java:197) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.action.ActionRunnable$2.doRun(ActionRunnable.java:86) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:50) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.common.util.concurrent.OpenSearchExecutors$DirectExecutorService.execute(OpenSearchExecutors.java:296) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.doStart(TransportMasterNodeAction.java:197) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.action.support.master.TransportMasterNodeAction.doExecute(TransportMasterNodeAction.java:140) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.action.support.master.TransportMasterNodeAction.doExecute(TransportMasterNodeAction.java:72) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:194) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.action.support.TransportAction.execute(TransportAction.java:169) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.action.support.TransportAction.execute(TransportAction.java:97) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.client.node.NodeClient.executeLocally(NodeClient.java:108) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.client.node.NodeClient.doExecute(NodeClient.java:95) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.client.support.AbstractClient.execute(AbstractClient.java:433) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.client.support.AbstractClient$ClusterAdmin.execute(AbstractClient.java:730) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.client.support.AbstractClient$ClusterAdmin.putRepository(AbstractClient.java:973) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.rest.action.admin.cluster.RestPutRepositoryAction.lambda$prepareRequest$0(RestPutRepositoryAction.java:75) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.rest.BaseRestHandler.handleRequest(BaseRestHandler.java:128) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.rest.RestController.dispatchRequest(RestController.java:306) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.rest.RestController.tryAllHandlers(RestController.java:392) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.rest.RestController.dispatchRequest(RestController.java:235) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.http.AbstractHttpServerTransport.dispatchRequest(AbstractHttpServerTransport.java:361) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.http.AbstractHttpServerTransport.handleIncomingRequest(AbstractHttpServerTransport.java:440) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.http.AbstractHttpServerTransport.incomingRequest(AbstractHttpServerTransport.java:351) [opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.http.netty4.Netty4HttpRequestHandler.channelRead0(Netty4HttpRequestHandler.java:55) [transport-netty4-client-1.2.4.jar:1.2.4]",
"at org.opensearch.http.netty4.Netty4HttpRequestHandler.channelRead0(Netty4HttpRequestHandler.java:41) [transport-netty4-client-1.2.4.jar:1.2.4]",
"at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at org.opensearch.http.netty4.Netty4HttpPipeliningHandler.channelRead(Netty4HttpPipeliningHandler.java:71) [transport-netty4-client-1.2.4.jar:1.2.4]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) [netty-codec-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) [netty-codec-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.handler.codec.MessageToMessageCodec.channelRead(MessageToMessageCodec.java:111) [netty-codec-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) [netty-codec-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) [netty-codec-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:324) [netty-codec-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296) [netty-codec-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286) [netty-handler-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) [netty-codec-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:722) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:623) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:586) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496) [netty-transport-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986) [netty-common-4.1.72.Final.jar:4.1.72.Final]",
"at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.72.Final.jar:4.1.72.Final]",
"at java.lang.Thread.run(Thread.java:834) [?:?]",
"Caused by: java.lang.IllegalStateException: Client instance has been closed.",
"at org.glassfish.jersey.internal.guava.Preconditions.checkState(Preconditions.java:169) ~[?:?]",
"at org.glassfish.jersey.client.JerseyClient.checkNotClosed(JerseyClient.java:257) ~[?:?]",
"at org.glassfish.jersey.client.JerseyClient.target(JerseyClient.java:272) ~[?:?]",
"at org.glassfish.jersey.client.JerseyClient.target(JerseyClient.java:56) ~[?:?]",
"at com.oracle.bmc.auth.AbstractFederationClientAuthenticationDetailsProviderBuilder.simpleRetry(AbstractFederationClientAuthenticationDetailsProviderBuilder.java:454) ~[?:?]",
"at com.oracle.bmc.auth.AbstractFederationClientAuthenticationDetailsProviderBuilder.autoDetectEndpointUsingMetadataUrl(AbstractFederationClientAuthenticationDetailsProviderBuilder.java:247) ~[?:?]",
"at com.oracle.bmc.auth.AbstractFederationClientAuthenticationDetailsProviderBuilder.autoDetectUsingMetadataUrl(AbstractFederationClientAuthenticationDetailsProviderBuilder.java:233) ~[?:?]",
"at com.oracle.bmc.auth.InstancePrincipalsAuthenticationDetailsProvider$InstancePrincipalsAuthenticationDetailsProviderBuilder.build(InstancePrincipalsAuthenticationDetailsProvider.java:77) ~[?:?]",
"at org.opensearch.repositories.oci.OciObjectStorageClientSettings.toAuthDetailsProvider(OciObjectStorageClientSettings.java:144) ~[?:?]",
"at org.opensearch.repositories.oci.OciObjectStorageClientSettings.<init>(OciObjectStorageClientSettings.java:112) ~[?:?]",
"at org.opensearch.repositories.oci.OciObjectStorageRepository.<init>(OciObjectStorageRepository.java:116) ~[?:?]",
"at org.opensearch.repositories.oci.OciObjectStoragePlugin.lambda$getRepositories$0(OciObjectStoragePlugin.java:54) ~[?:?]",
"at org.opensearch.repositories.Repository$Factory.create(Repository.java:88) ~[opensearch-1.2.4.jar:1.2.4]",
"at org.opensearch.repositories.RepositoriesService.createRepository(RepositoriesService.java:563) ~[opensearch-1.2.4.jar:1.2.4]",
"... 79 more"] }

After fixing the plugin logging I am not down to this root exception as cause:

"Caused by: org.opensearch.transport.NodeDisconnectedException: [opensearch-es-master-2][10.244.1.113:9300][cluster:monitor/health] disconnected"] }
{"type": "server", "timestamp": "2022-12-02T17:19:25,338Z", "level": "INFO", "component": "o.o.r.o.OciObjectStorageClientSettings", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "Initializing client using instance principals", "cluster.uuid": "lIIEkujLQDON2n7gTokYiw", "node.id": "BI3S8Y57TPa7Mk5ucKJHYQ"  }
{"type": "server", "timestamp": "2022-12-02T17:19:25,358Z", "level": "WARN", "component": "stderr", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "[opensearch[opensearch-es-master-0][transport_worker][T#1]] INFO com.oracle.bmc.Services - Registering new service: Services.BasicService(serviceName=AUTH, serviceEndpointPrefix=auth, serviceEndpointTemplate=null)", "cluster.uuid": "lIIEkujLQDON2n7gTokYiw", "node.id": "BI3S8Y57TPa7Mk5ucKJHYQ"  }
{"type": "server", "timestamp": "2022-12-02T17:19:27,044Z", "level": "WARN", "component": "stderr", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "[opensearch[opensearch-es-master-0][transport_worker][T#1]] WARN com.oracle.bmc.auth.AbstractFederationClientAuthenticationDetailsProviderBuilder - Attempt 1 - Rest call to get region from metadata service failed ", "cluster.uuid": "lIIEkujLQDON2n7gTokYiw", "node.id": "BI3S8Y57TPa7Mk5ucKJHYQ"  }
{"type": "server", "timestamp": "2022-12-02T17:19:27,045Z", "level": "WARN", "component": "stderr", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "javax.ws.rs.ProcessingException: java.security.AccessControlException: access denied (\"java.net.SocketPermission\" \"169.254.169.254:80\" \"connect,resolve\")", "cluster.uuid": "lIIEkujLQDON2n7gTokYiw", "node.id": "BI3S8Y57TPa7Mk5ucKJHYQ"  }
{"type": "server", "timestamp": "2022-12-02T17:19:27,046Z", "level": "WARN", "component": "stderr", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "\tat org.glassfish.jersey.client.ClientRuntime.invoke(ClientRuntime.java:309)", "cluster.uuid": "lIIEkujLQDON2n7gTokYiw", "node.id": "BI3S8Y57TPa7Mk5ucKJHYQ"  }
{"type": "server", "timestamp": "2022-12-02T17:19:27,047Z", "level": "WARN", "component": "stderr", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "\tat org.glassfish.jersey.client.JerseyInvocation.lambda$invoke$0(JerseyInvocation.java:662)", "cluster.uuid": "lIIEkujLQDON2n7gTokYiw", "node.id": "BI3S8Y57TPa7Mk5ucKJHYQ"  }
{"type": "server", "timestamp": "2022-12-02T17:19:27,047Z", "level": "WARN", "component": "stderr", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "\tat org.glassfish.jersey.client.JerseyInvocation.call(JerseyInvocation.java:697)", "cluster.uuid": "lIIEkujLQDON2n7gTokYiw", "node.id": "BI3S8Y57TPa7Mk5ucKJHYQ"  }
{"type": "server", "timestamp": "2022-12-02T17:19:27,047Z", "level": "WARN", "component": "stderr", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "\tat org.glassfish.jersey.client.JerseyInvocation.lambda$runInScope$3(JerseyInvocation.java:691)", "cluster.uuid": "lIIEkujLQDON2n7gTokYiw", "node.id": "BI3S8Y57TPa7Mk5ucKJHYQ"  }
{"type": "server", "timestamp": "2022-12-02T17:19:27,047Z", "level": "WARN", "component": "stderr", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "\tat org.glassfish.jersey.internal.Errors.process(Errors.java:292)", "cluster.uuid": "lIIEkujLQDON2n7gTokYiw", "node.id": "BI3S8Y57TPa7Mk5ucKJHYQ"  }
{"type": "server", "timestamp": "2022-12-02T17:19:27,048Z", "level": "WARN", "component": "stderr", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "\tat org.glassfish.jersey.internal.Errors.process(Errors.java:274)", "cluster.uuid": "lIIEkujLQDON2n7gTokYiw", "node.id": "BI3S8Y57TPa7Mk5ucKJHYQ"  }
{"type": "server", "timestamp": "2022-12-02T17:19:27,048Z", "level": "WARN", "component": "stderr", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "\tat org.glassfish.jersey.internal.Errors.process(Errors.java:205)", "cluster.uuid": "lIIEkujLQDON2n7gTokYiw", "node.id": "BI3S8Y57TPa7Mk5ucKJHYQ"  }
{"type": "server", "timestamp": "2022-12-02T17:19:27,048Z", "level": "WARN", "component": "stderr", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "\tat org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:390)", "cluster.uuid": "lIIEkujLQDON2n7gTokYiw", "node.id": "BI3S8Y57TPa7Mk5ucKJHYQ"  }
{"type": "server", "timestamp": "2022-12-02T17:19:27,049Z", "level": "WARN", "component": "stderr", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "\tat org.glassfish.jersey.client.JerseyInvocation.runInScope(JerseyInvocation.java:691)", "cluster.uuid": "lIIEkujLQDON2n7gTokYiw", "node.id": "BI3S8Y57TPa7Mk5ucKJHYQ"  }
{"type": "server", "timestamp": "2022-12-02T17:19:27,049Z", "level": "WARN", "component": "stderr", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "\tat org.glassfish.jersey.client.JerseyInvocation.invoke(JerseyInvocation.java:661)", "cluster.uuid": "lIIEkujLQDON2n7gTokYiw", "node.id": "BI3S8Y57TPa7Mk5ucKJHYQ"  }
{"type": "server", "timestamp": "2022-12-02T17:19:27,049Z", "level": "WARN", "component": "stderr", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "\tat org.glassfish.jersey.client.JerseyInvocation$Builder.method(JerseyInvocation.java:413)", "cluster.uuid": "lIIEkujLQDON2n7gTokYiw", "node.id": "BI3S8Y57TPa7Mk5ucKJHYQ"  }
{"type": "server", "timestamp": "2022-12-02T17:19:27,049Z", "level": "WARN", "component": "stderr", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "\tat org.glassfish.jersey.client.JerseyInvocation$Builder.get(JerseyInvocation.java:313)", "cluster.uuid": "lIIEkujLQDON2n7gTokYiw", "node.id": "BI3S8Y57TPa7Mk5ucKJHYQ"  }
{"type": "server", "timestamp": "2022-12-02T17:19:27,050Z", "level": "WARN", "component": "stderr", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "\tat com.oracle.bmc.auth.AbstractFederationClientAuthenticationDetailsProviderBuilder.lambda$executeInstanceFallback$1(AbstractFederationClientAuthenticationDetailsProviderBuilder.java:346)", "cluster.uuid": "lIIEkujLQDON2n7gTokYiw", "node.id": "BI3S8Y57TPa7Mk5ucKJHYQ"  }
{"type": "server", "timestamp": "2022-12-02T17:19:27,050Z", "level": "WARN", "component": "stderr", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "\tat com.oracle.bmc.auth.AbstractFederationClientAuthenticationDetailsProviderBuilder.simpleRetry(AbstractFederationClientAuthenticationDetailsProviderBuilder.java:455)", "cluster.uuid": "lIIEkujLQDON2n7gTokYiw", "node.id": "BI3S8Y57TPa7Mk5ucKJHYQ"  }
{"type": "server", "timestamp": "2022-12-02T17:19:27,050Z", "level": "WARN", "component": "stderr", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "\tat com.oracle.bmc.auth.AbstractFederationClientAuthenticationDetailsProviderBuilder.executeInstanceFallback(AbstractFederationClientAuthenticationDetailsProviderBuilder.java:338)", "cluster.uuid": "lIIEkujLQDON2n7gTokYiw", "node.id": "BI3S8Y57TPa7Mk5ucKJHYQ"  }
{"type": "server", "timestamp": "2022-12-02T17:19:27,050Z", "level": "WARN", "component": "stderr", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "\tat com.oracle.bmc.auth.AbstractFederationClientAuthenticationDetailsProviderBuilder.autoDetectEndpointUsingMetadataUrl(AbstractFederationClientAuthenticationDetailsProviderBuilder.java:245)", "cluster.uuid": "lIIEkujLQDON2n7gTokYiw", "node.id": "BI3S8Y57TPa7Mk5ucKJHYQ"  }
{"type": "server", "timestamp": "2022-12-02T17:19:27,051Z", "level": "WARN", "component": "stderr", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "\tat com.oracle.bmc.auth.AbstractFederationClientAuthenticationDetailsProviderBuilder.autoDetectUsingMetadataUrl(AbstractFederationClientAuthenticationDetailsProviderBuilder.java:233)", "cluster.uuid": "lIIEkujLQDON2n7gTokYiw", "node.id": "BI3S8Y57TPa7Mk5ucKJHYQ"  }
{"type": "server", "timestamp": "2022-12-02T17:19:27,051Z", "level": "WARN", "component": "stderr", "cluster.name": "opensearch-tracking", "node.name": "opensearch-es-master-0", "message": "\tat com.oracle.bmc.auth.InstancePrincipalsAuthenticationDetailsProvider$InstancePrincipalsAuthenticationDetailsProviderBuilder.build(InstancePrincipalsAuthenticationDetailsProvider.java:77)"...

--- a/oci-repository-plugin/pom.xml
+++ b/oci-repository-plugin/pom.xml
@@ -25,7 +25,15 @@
 
     </properties>
 
-  <dependencies>
+    <dependencies>
+
+      <!-- Required so that we can get logging from OCI SDK  -->
+      <dependency>
+          <groupId>org.slf4j</groupId>
+          <artifactId>slf4j-simple</artifactId>
+          <version>${slf4j.version}</version>
+      </dependency>
+      
       <!-- Cross project dependencies-->
       <dependency>
           <groupId>org.opensearch.fixtures.oci</groupId>

--- a/oci-repository-plugin/src/main/java/org/opensearch/repositories/oci/OciObjectStorageClientSettings.java
+++ b/oci-repository-plugin/src/main/java/org/opensearch/repositories/oci/OciObjectStorageClientSettings.java
@@ -23,6 +23,11 @@ import org.opensearch.common.settings.Setting;
 import java.io.FileInputStream;
 import java.io.InputStream;
 
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.security.Permission;
+import java.security.AllPermission;
+
 import static org.opensearch.common.settings.Setting.boolSetting;
 import static org.opensearch.common.settings.Setting.simpleString;
 
@@ -141,6 +146,39 @@ public class OciObjectStorageClientSettings {
     }
 
     private static BasicAuthenticationDetailsProvider toAuthDetailsProvider() {
-        return InstancePrincipalsAuthenticationDetailsProvider.builder().build();
+
+        /*
+         * Work around security bug while using Instance Principal Authentication
+         * where adding the required java.net.SocketPermission to oci-repository-plugin/src/main/resources/plugin-security.policy
+         * doesn't fix the issue. See https://github.com/opensearch-project/opensearch-oci-object-storage/issues/13
+         */
+        SecurityManager sm = System.getSecurityManager();
+
+        try {
+
+            AccessController.doPrivileged(
+                    (PrivilegedAction<Object>)
+                            () -> {
+                                System.setSecurityManager(
+                                        new SecurityManager() {
+
+                                            @Override
+                                            public void checkPermission(Permission perm) {
+                                                if (perm instanceof AllPermission) {
+                                                    throw new SecurityException();
+                                                }
+                                            }
+                                        });
+                                return null;
+                            });
+
+            return InstancePrincipalsAuthenticationDetailsProvider.builder().build();
+
+        } catch (Exception ex) {
+            log.error("Failure calling toAuthDetailsProvider", ex);
+            throw ex;
+        } finally {
+            System.setSecurityManager(sm);
+        }
     }
 }

--- a/pom.xml
+++ b/pom.xml
@@ -20,6 +20,7 @@
         <management-plane-version>0.2.143</management-plane-version>
         <jersey-version>2.25.1</jersey-version>
         <jackson-version>2.12.5</jackson-version>
+	<slf4j.version>1.7.21</slf4j.version>
     </properties>
 
     <modules>
@@ -27,7 +28,15 @@
         <module>oci-repository-plugin</module>
     </modules>
     <dependencyManagement>
-        <dependencies>
+      <dependencies>
+
+	    <!-- Required so that we can get logging from OCI SDK  -->
+	    <dependency>
+                <groupId>org.slf4j</groupId>
+                <artifactId>slf4j-simple</artifactId>
+                <version>${slf4j.version}</version>
+            </dependency>
+	    
             <!-- Cross project dependencies-->
             <dependency>
                 <groupId>com.oracle.pic.opensearch</groupId>

